tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,392 @@
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(2) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var oracleDbName = "freepdb1";
+
+        var oracle = builder.AddOracle("oracle");
+        var db = oracle.AddDatabase(oracleDbName);
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddOracleDatabaseDbContext<TestDbContext>(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        // Wait until the database is available
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+            Assert.True(await databaseCreator.CanConnectAsync(token));
+        }, cts.Token);
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+
+            // Don't dispose the connection as it is managed by EF Core
+            var connection = dbContext.Database.GetDbConnection();
+
+            if (connection.State != System.Data.ConnectionState.Open)
+            {
+                await connection.OpenAsync(token);
+            }
+
+            var command = connection.CreateCommand();
+            command.CommandText = "SELECT 1";
+            var results = await command.ExecuteReaderAsync(token);
+
+            Assert.True(results.HasRows);
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false, Skip = "When using p@ssw0rd1 it fails with: Password cannot be null. Enter password: \n Then when using a different password the image takes 5 minutes to be ready.")]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+        var oracleDbName = "freepdb1";
+
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(2) })
+            .Build();
+
+        try
+        {
+            var builder1 = CreateDistributedApplicationBuilder();
+
+            // Prevent CredScan warnings
+            var password = PasswordGenerator.Generate(8, true, true, true, false, 1, 1, 1, 0);
```

Comment:
How does this prevent credscan warnings?

Reply:
There is no hard-coded value so CredScan won't trigger. You have to assign something at some point anyway so it's not about it that CredScan would complain, it's what you assign to it. It's just to prevent actual values from being leaked. At least that how I believe it works.